backend/tests/integration-async/records/api/test_base.py

```diff
@@ -140,3 +138,57 @@ def test_get_modified_references_returns_no_references_when_non_impacting_metada
     db.session.commit()
 
     assert citing_record.get_modified_references() == []
+
+
+def test_revert_revision_works_correctly_and_runs_update(inspire_app):
```

Comment:
Shall we test also that the data is correct after the revert?

Reply:
Check is already there (check line 194)